src/components/gating/GatingSubscriber.js

```diff
@@ -0,0 +1,257 @@
+import React, { useMemo } from 'react';
+import TextField from '@material-ui/core/TextField';
+import BaseScatterplotSubscriber, {
+  BASE_SCATTERPLOT_DATA_TYPES,
+} from '../base-scatterplot/BaseScatterplotSubscriber';
+import { capitalize } from '../../utils';
+import {
+  useLoaders,
+  useCoordination,
+} from '../../app/state/hooks';
+import {
+  useReady, useUrls,
+} from '../hooks';
+import {
+  useCellsData,
+  useExpressionMatrixData,
+} from '../data-hooks';
+import OptionSelect from '../shared-plot-options/OptionSelect';
+import { useStyles } from '../shared-plot-options/styles';
+import { COMPONENT_COORDINATION_TYPES } from '../../app/state/coordination';
+
+
+/**
+   * A subscriber component for the gating scatterplot.
+   * @param {object} props
+   * @param {number} props.uuid The unique identifier for this component.
+   * @param {string} props.theme The current theme name.
+   * @param {object} props.coordinationScopes The mapping from coordination types to coordination
+   * scopes.
+   * @param {boolean} props.disableTooltip Should the tooltip be disabled?
+   * @param {function} props.removeGridComponent The callback function to pass to TitleInfo,
+   * to call when the component has been removed from the grid.
+   * @param {number} props.averageFillDensity Override the average fill density calculation
+   * when using dynamic opacity mode.
+   */
+export default function GatingSubscriber(props) {
+  const {
+    coordinationScopes,
+    observationsLabelOverride: observationsLabel = 'cell',
+  } = props;
+
+  // Get "props" from the coordination space.
+  const [{ dataset }] = useCoordination(
+    COMPONENT_COORDINATION_TYPES.scatterplot,
+    coordinationScopes,
+  );
+
+  // Get data from loaders using the data hooks.
+  const [urls, addUrl, resetUrls] = useUrls();
+  const loaders = useLoaders();
+  const [isReady, setItemIsReady, setItemIsNotReady, resetReadyItems] = useReady(
+    BASE_SCATTERPLOT_DATA_TYPES,
+  );
+  const [cells, cellsCount] = useCellsData(loaders, dataset, setItemIsReady, addUrl, true);
+  const [expressionMatrix] = useExpressionMatrixData(
+    loaders, dataset, setItemIsReady, addUrl, true,
+  );
+
+  // State for the custom setting options.
+  const [selectedGenes, setSelectedGenes] = React.useState([]);
+  const [transformType, setTransformType] = React.useState('');
+  const [transformCoefficient, setTransformCoefficient] = React.useState('');
```

Comment:
Our goal is to manage these types of state via the "coordination model", allowing values to be read/written via the view config (incl. in our Python and R widget wrappers) and linked across multiple views. For that reason, I would prefer to re-use existing coordination types or create new coordination types if none of the existing ones are applicable.

For example, we have the coordination type [`featureValueTransform`](https://github.com/vitessce/vitessce/blob/main/src/schemas/config-1.0.13.schema.json#L1012) which can take the value `log1p` but we can certainly add `log` and `arcsinh` to the schema as well, if this would make sense.

We do not have any existing coordination type like `transformCoefficient` so feel free to create one, maybe `featureValueTransformCoefficient` to be consistent with the other names.

The [`featureSelection`](https://github.com/vitessce/vitessce/blob/main/src/schemas/config-1.0.13.schema.json#L999) (formerly `geneSelection`) coordination type can take an array of gene IDs.

The prop [`enableMultiSelect`](https://github.com/vitessce/vitessce/blob/keller-mark/fom-file-types/src/components/genes/GenesSubscriber.js#L37) facilitates selecting multiple genes with shift+click in the gene list view, and can be set in the config like

```js
{
  component: 'genes',
  props: {
    enableMultiSelect: true,
  },
  x: 9, y: 0, w: 3, h: 2,
}
```

I think we should also implement a prop like `maxMultiSelect` that is `null` by default, but we can set to `2` when used alongside the gating component. When it is not `null`, we can show selection order to the user, for instance if `Aldoc` is selected first and `Bmp4` second we would display indicators like
![proposed gene list selection order indicators](https://user-images.githubusercontent.com/7525285/179761989-c39272b3-b8cc-4fcb-8345-fa611c661f3a.png)

I see you also implemented a gene selection interface in the Options dropdown menu, which makes sense since it may be a waste of screen real estate to render the `genes` view as a separate grid element. However, we should not render the gene selection interface in the Options by default, and instead allow opting-in with a prop like `enableFeatureSelectionOption` (since scRNA-seq data can contain 20,000 genes which we handle via [react-virtualized](https://github.com/bvaughn/react-virtualized) in the `genes` view).

Reply:
Ah I see. Then in this PR we can ignore the `genes` view. (Maybe we should eventually implement a more complex interface that allows users to assign each gene selection to one of (X, Y, color), but that feels out of scope for this PR, so I will make a new issue to discuss it.)

I think we can work around the issue by creating two new coordination types, 
`gatingFeatureSelectionX` and `gatingFeatureSelectionY` which can each take one gene ID. Then we can use the current `featureSelection` coordination and `genes` view behaviors to do the coloring. Does that make sense?

Reply:
That makes sense! Can do.